test/integration/concierge_jwtauthenticator_status_test.go

```diff
@@ -0,0 +1,297 @@
+							},
+						},
+					))
+			},
+		},
+	}
+	for _, test := range tests {
+		tt := test
+		t.Run(tt.name, func(t *testing.T) {
+			tt.run(t)
+		})
+	}
+}
+
+func TestConciergeJWTAuthenticatorCRDValidations_Parallel(t *testing.T) {
+	jwtAuthenticatorClient := testlib.NewConciergeClientset(t).AuthenticationV1alpha1().JWTAuthenticators()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	t.Cleanup(cancel)
+
+	objectMeta := testlib.ObjectMetaWithRandomName(t, "jwt-authenticator")
+	tests := []struct {
+		name             string
+		jwtAuthenticator *v1alpha1.JWTAuthenticator
+		wantErr          string
+		// some tests change the environment (api group suffix pinniped.dev->walrus.tld) so
+		// we need to be able to compare against several error strings
+		wantOneOfErr []string
+	}{
+		{
+			name: "issuer can not be empty string",
+			jwtAuthenticator: &v1alpha1.JWTAuthenticator{
+				ObjectMeta: objectMeta,
+				Spec: v1alpha1.JWTAuthenticatorSpec{
+					Issuer:   "",
+					Audience: "fake-audience",
+				},
+			},
+			wantOneOfErr: []string{
+				`JWTAuthenticator.authentication.concierge.pinniped.dev "` + objectMeta.Name + `" is invalid: ` +
+					`spec.issuer: Invalid value: "": spec.issuer in body should be at least 1 chars long`,
+				`JWTAuthenticator.authentication.concierge.walrus.tld "` + objectMeta.Name + `" is invalid: ` +
+					`spec.issuer: Invalid value: "": spec.issuer in body should be at least 1 chars long`,
+			},
+		},
+		{
+			name: "audience can not be empty string",
+			jwtAuthenticator: &v1alpha1.JWTAuthenticator{
+				ObjectMeta: objectMeta,
+				Spec: v1alpha1.JWTAuthenticatorSpec{
+					Issuer:   "https://example.com",
+					Audience: "",
+				},
+			},
+			wantOneOfErr: []string{
+				`JWTAuthenticator.authentication.concierge.pinniped.dev "` + objectMeta.Name + `" is invalid: ` +
+					`spec.audience: Invalid value: "": spec.audience in body should be at least 1 chars long`,
+				`JWTAuthenticator.authentication.concierge.walrus.tld "` + objectMeta.Name + `" is invalid: ` +
+					`spec.audience: Invalid value: "": spec.audience in body should be at least 1 chars long`,
+			},
+		},
+		{
+			name: "issuer must be https",
+			jwtAuthenticator: &v1alpha1.JWTAuthenticator{
+				ObjectMeta: objectMeta,
+				Spec: v1alpha1.JWTAuthenticatorSpec{
+					Issuer:   "http://www.example.com",
+					Audience: "foo",
+				},
+			},
+			wantOneOfErr: []string{
+				`JWTAuthenticator.authentication.concierge.pinniped.dev "` + objectMeta.Name + `" is invalid: ` +
+					`spec.issuer: Invalid value: "http://www.example.com": spec.issuer in body should match '^https://'`,
+				`JWTAuthenticator.authentication.concierge.walrus.tld "` + objectMeta.Name + `" is invalid: ` +
+					`spec.issuer: Invalid value: "http://www.example.com": spec.issuer in body should match '^https://'`,
+			},
+		},
+		{
+			name: "minimum valid authenticator",
```

Comment:
naming nit: This test is the only one that is the "minimum valid authenticator". The other tests below maybe should not say "minimum valid authenticator" since they go beyond what is the minimum.

Reply:
For all the valid tests, isn't the code really going to perform discovery against `https://www.example.com/.well-known/openid-configuration`? That is a real endpoint on the internet that will yield a real response.

Perhaps you could instead use `env.CLIUpstreamOIDC.Issuer` which is a real OIDC issuer.

Reply:
These are CRD validations so they won't perform any actual checks, but I'll update this anyway.